### ISSUE TITLE
Window calico node unable to create hns network calico subnet

### ISSUE THREAD

**Issue by bluefriday** [projectcalico/calico#4533]
## Expected Behavior
I'm testing about running calico with k8s on 'Windows 2019' node.

## Current Behavior
With 'vxlan' option, I ran powershell script running calico, but calico could not create subnet.
```
2021-04-15 15:55:53.604 [INFO][46540] startup/ipam.go 1952: Looking up existing affinities for host host="win-node02"
2021-04-15 15:55:53.607 [INFO][46540] startup/ipam.go 349: Looking up existing affinities for host host="win-node02"
2021-04-15 15:55:53.609 [INFO][46540] startup/ipam.go 432: Trying affinity for 10.35.64.0/26 host="win-node02"
2021-04-15 15:55:53.611 [INFO][46540] startup/ipam.go 143: Attempting to load block cidr=10.35.64.0/26 host="win-node02"
2021-04-15 15:55:53.613 [INFO][46540] startup/ipam.go 220: Affinity is confirmed and block has been loaded cidr=10.35.64.0/26 host="win-node02"
2021-04-15 15:55:53.613 [INFO][46540] startup/ipam.go 1978: Host's block '10.35.64.0/26'  host="win-node02"
2021-04-15 15:55:53.622 [INFO][46540] startup/dataplane_windows.go 294: Found existing HNS network [&{Id:5C266810-4DA8-4BF2-1234-2EF76A4E3BD4 Name:Calico Type:Overlay NetworkAdapterName: SourceMac: Policies:[] MacPools:[{StartMacAddress:00-15-5D-14-50-00 EndMacAddress:00-15-5D-14-5F-FF}] Subnets:[{AddressPrefix:10.35.64.0/26 GatewayAddress:10.35.64.1 Policies:[[123 34 84 121 112 101 34 58 34 86 83 73 68 34 44 34 86 83 73 68 34 58 52 48 57 54 125]]}] DNSSuffix: DNSServerList: DNSServerCompartment:3 ManagementIP:172.18.10.145 AutomaticDNS:false}] subnet="10.35.64.0/26"
2021-04-15 15:55:53.623 [ERROR][46540] startup/dataplane_windows.go 117: Unable to create hns network Calico subnet="10.35.64.0/26"
2021-04-15 15:55:53.623 [ERROR][46540] startup/startup.go 224: Unable to ensure network for os error=Could not find vxlan0 in V2: Network ID "5C266810-4DA8-4BF2-1234-2EF76A4E3BD4" not found
2021-04-15 15:55:53.623 [WARNING][46540] startup/startup.go 1340: Terminating
Calico node initialisation failed, will retry...
```

## Possible Solution
I want know how to fix this.
Thank you 👍 

## Steps to Reproduce (for bugs)
I used following guides.
- https://docs.projectcalico.org/getting-started/kubernetes/self-managed-onprem/onpremises

## Your Environment
* Calico version : v3.17.3
* Orchestrator version (e.g. kubernetes, mesos, rkt): kubernetes v1.19.6
* Operating System and version: windows server 2019 (ver 1809)

**Comment by bluefriday** [comment]
I used contained instead of Docker. And it's my calico config
```

## Cluster configuration:

# KUBE_NETWORK should be set to a regular expression that matches the HNS network(s) used for pods.
# The default, "Calico.*", is correct for Calico CNI. 
$env:KUBE_NETWORK = "calico.*"

# Set this to one of the following values:
# - "vxlan" for Calico VXLAN networking
# - "windows-bgp" for Calico BGP networking using the Windows BGP router.
# - "none" to disable the Calico CNI plugin (so that you can use another plugin).
$env:CALICO_NETWORKING_BACKEND="vxlan"

# Set to match your Kubernetes service CIDR.
$env:K8S_SERVICE_CIDR = "10.96.0.0/12"
$env:DNS_NAME_SERVERS = "10.96.0.10"
$env:DNS_SEARCH = "svc.cluster.local"


## Datastore configuration:

# Set this to "kubernetes" to use the kubernetes datastore, or "kubernetes" for etcd.
$env:CALICO_DATASTORE_TYPE = "kubernetes"

# Set KUBECONFIG to the path of your kubeconfig file.
#$env:KUBECONFIG = "$PSScriptRoot\calico-kube-config"
$env:KUBECONFIG = "$PSScriptRoot\config"

# For the "kubernetes" datastore only: set ETCD_ENDPOINTS, format: "http://<host>:<port>,..."
$env:ETCD_ENDPOINTS = "[APISERVER_IP]"
# For etcd over TLS, set these lines to point to your keys/certs:
$env:ETCD_KEY_FILE = "c:\CalicoWindows\etcd-tls\server.key"
$env:ETCD_CERT_FILE = "c:\CalicoWindows\etcd-tls\server.crt"
$env:ETCD_CA_CERT_FILE = "c:\CalicoWindows\etcd-tls\ca.crt"


## CNI configuration, only used for the "vxlan" networking backends.

# Place to install the CNI plugin to.  Should match kubelet's --cni-bin-dir.
$env:CNI_BIN_DIR = "C:\opt\cni\bin"
# Place to install the CNI config to.  Should be located in kubelet's --cni-conf-dir.
$env:CNI_CONF_DIR = "C:\etc\cni\net.d"
$env:CNI_CONF_FILENAME = "10-calico.conf"
# IPAM type to use with Calico's CNI plugin.  One of "calico-ipam" or "host-local".
$env:CNI_IPAM_TYPE = "calico-ipam"
$env:CALICO_IPV4POOL_CIDR = "10.32.0.0/12"

## VXLAN-specific configuration.

# The VXLAN VNI / VSID.  Must match the VXLANVNI felix configuration parameter used
# for Linux nodes.
$env:VXLAN_VNI = "4096"
# Prefix used when generating MAC addresses for virtual NICs.
$env:VXLAN_MAC_PREFIX = "0E-2A"


## Node configuration.

# The NODENAME variable should be set to match the Kubernetes Node name of this host.
# The default uses this node's hostname (which is the same as kubelet).
#
# Note: on AWS, kubelet is often configured to use the internal domain name of the host rather than
# the simple hostname, for example "ip-172-16-101-135.us-west-2.compute.internal".
$env:NODENAME = $(hostname).ToLower()
# Similarly, CALICO_K8S_NODE_REF should be set to the Kubernetes Node name.  When using etcd,
# the Calico kube-controllers pod will clean up Calico node objects if the corresponding Kubernetes Node is
# cleaned up.
$env:CALICO_K8S_NODE_REF = $env:NODENAME

# The time out to wait for a valid IP of an interface to be assigned before initialising Calico
# after a reboot.
$env:STARTUP_VALID_IP_TIMEOUT = 90

# The IP of the node; the default will auto-detect a usable IP in most cases.
$env:IP = "autodetect"

## Logging.

$env:CALICO_LOG_DIR = "$PSScriptRoot\logs"

# Felix logs to screen at info level by default.  Uncomment this line to override the log
# level.  Alternatively, (if this is commented out) the log level can be controlled via
# the FelixConfiguration resource in the datastore.
# $env:FELIX_LOGSEVERITYSCREEN = "info"
# Disable logging to file by default since the service wrapper will redirect our log to file.
$env:FELIX_LOGSEVERITYFILE = "none"
# Disable syslog logging, which is not supported on Windows.
$env:FELIX_LOGSEVERITYSYS = "none"
# confd logs to screen at info level by default.  Uncomment this line to override the log
# level.
#$env:BGP_LOGSEVERITYSCREEN = "debug"
```

**Comment by lwr20** [comment]
@song-jiang @lmm will know better than I do, but I think containerd isn't supported yet.

**Comment by lmm** [comment]
Thanks @bluefriday - @lwr20 is spot on, we're working on containerd support which should be in an upcoming release.

That said, the errors you're seeing are strange. The HNS network is found and then a short while later it's not found. Let me look into this and get back to you.

Could you try the installation with Docker installed to see if that works for you with your setup?

**Comment by lwr20** [comment]
This work is now available in Calico v3.19.0
https://docs.projectcalico.org/release-notes/#windows-data-plane-support-for-containerd
